lib/nolibc/stdio.c

```diff
@@ -541,3 +543,78 @@ int puts(const char *s)
 {
 	return fputs_internal(s, stdout, 1);
 }
+
+#if CONFIG_LIBVFSCORE
+
+int __fmodeflags(const char *mode)
+{
+	int flags;
+
+	if (strchr(mode, '+'))
+		flags = O_RDWR;
+	else if (*mode == 'r')
+		flags = O_RDONLY;
+	else
+		flags = O_WRONLY;
+	if (strchr(mode, 'x'))
+		flags |= O_EXCL;
+	if (strchr(mode, 'e'))
+		flags |= O_CLOEXEC;
+	if (*mode != 'r')
+		flags |= O_CREAT;
+	if (*mode == 'w')
+		flags |= O_TRUNC;
+	if (*mode == 'a')
+		flags |= O_APPEND;
+	return flags;
+}
+
+struct vfscore_file *fopen(const char *pathname, const char *mode)
+{
+	struct vfscore_file *f;
+	int flags;
+	int fd;
+
+	if (!strchr("rwa", *mode)) {
+		errno = EINVAL;
+		return 0;
+	}
+
+	flags = __fmodeflags(mode);
+	fd = open(pathname, flags, 0666);
+
+	return vfscore_get_file(fd);
+}
+
+int feof(struct vfscore_file *stream)
+{
+	return 0;
+}
+
+int fclose(struct vfscore_file *stream)
+{
+	return close(stream->fd);
+}
+
+struct vfscore_file *fdopen(int fd, const char *mode)
+{
+	return vfscore_get_file(fd);
+}
+
+int fseek(struct vfscore_file *stream, long offset, int whence)
+{
+	return lseek(stream->fd, offset, whence);
+}
+
+size_t fread(void *ptr, size_t size, size_t nmemb, struct vfscore_file *stream)
+{
```

Comment:
I think you should add the same check for `NULL` for `stream variable,  as you did for `fwrite()` to keep consistency.